frontend/src/app/components/network-graph/network-graph.component.ts

```diff
@@ -821,8 +832,38 @@ export class NetworkGraphComponent implements OnInit, OnChanges {
     this.infoBoxOverlay.selectAll('.info').remove();
   }
 
-  private selectNode(selectedNode: Node) {
+  private selectNode(selectedNode: SimulationNode) {
+    this.highlightNode(selectedNode);
+    this.highlightedNode = selectedNode;
+    this.drawNodeInfoBox();
+  }
+
+  private drawNodeInfoBox() {
+    if (!this.highlightedNode) {
+      return;
+    }
+    const selectedNode = this.highlightedNode;
+    const neighbors: Node[] = this.getNeighbors(selectedNode);
+
+    const info = this.nodeInfo(selectedNode);
+
+    const boxHeight = info.length * 15 + 10;
+    const boxWidth = 320;
+
+    const neighborY = (neighbors as SimulationNodeDatum[]).map(value => value.y).sort();
+
+    const boxX = this.getBoxX(selectedNode.x, selectedNode.x, boxWidth);
+
+    const minY = Math.min(selectedNode.y, neighborY[0]);
+    const maxY = Math.max(selectedNode.y, neighborY[neighborY.length - 1]);
+    const boxY = this.getBoxY(minY, maxY, 10, boxHeight);
+
+    this.drawInformationBox(boxX, boxY, boxWidth, info);
```

Comment:
I know this is just taken from the code-base as is and you can close this conversation right away, but I just want to say: I have no clue what this is doing.

Reply:
The whole network visualization is a bit too complex imho